packages/core/scripts/local/getGasEthHistoricalPrice.js

```diff
@@ -0,0 +1,103 @@
+ *
+ * @dev Prequisites: before running, you will need to set a GOOGLE_APPLICATIONS_CREDENTIALS environment variable for a service account.
+ * @dev This service account will need GCP admin or BigQuery permissions. This guide provides further instructions: https://cloud.google.com/docs/authentication/getting-started
+ * @dev How to run: yarn truffle exec ./scripts/local/getGasEthHistoricalPrice.js --network mainnet_mnemonic --time 1612137600
+ */
+const { BigQuery } = require("@google-cloud/bigquery");
+const highland = require("highland");
+const moment = require("moment");
+const { fromWei } = web3.utils;
+const argv = require("minimist")(process.argv.slice(), { string: ["identifier", "time"] });
+const { createQuery } = require("./GasEthQuery");
+
+const client = new BigQuery();
+const dateFormat = "YYYY-MM-DD HH:mm:ss";
+const precisionToUse = 18;
+
+async function getGasEthHistoricalPrice(callback) {
+  try {
+    // If user did not specify an identifier, provide a default value.
+    let queryIdentifier;
+    if (!argv.identifier) {
+      queryIdentifier = "GASETH-TWAP-1Mx1M";
+      console.log(`Optional '--identifier' flag not specified, defaulting to: ${queryIdentifier}`);
+    } else {
+      queryIdentifier = argv.identifier;
+    }
+
+    // Function to get the current time.
+    const getTime = () => Math.round(new Date().getTime() / 1000);
+
+    // If user specified a timestamp, then use it, otherwise default to the current time.
+    let queryTime;
+    if (!argv.time) {
+      queryTime = getTime();
+      console.log(
+        `Optional '--time' flag not specified, defaulting to the current Unix timestamp (in seconds): ${queryTime}`
+      );
+    } else {
+      queryTime = argv.time;
+    }
+    console.log(`⏰ Fetching nearest prices for the timestamp: ${new Date(queryTime * 1000).toUTCString()}`);
+
+    queryTime = new Date(queryTime * 1000);
+    // Using moment package to convert queryTime to a usable BQ UTC format.
+    let laterTimeBound = moment(queryTime)
+      .utc()
+      .format(dateFormat);
+
+    // Subtracting 30 days from the current time to give the earlier time bound.
+    let earlierTimeBound = moment(queryTime)
+      .subtract(30, "days")
+      .utc()
+      .format(dateFormat);
+
+    // Create the query with the needed time interval.
+    const query = createQuery(earlierTimeBound, laterTimeBound);
+    // Submit async call to BigQuery and check response.
+    let queryPrice;
+    try {
+      queryPrice = await runQuery(query);
+      queryPrice = queryPrice[0].gas_price;
+    } catch (err) {
+      callback(err);
+      return;
+    }
+
+    // Scaling by one million as specified in the GASETH-TWAP-1Mx1M UMIP
+    queryPrice = queryPrice * 1000000;
+
+    console.log(`\n⚠️ Truncating price to ${precisionToUse} decimals`);
+    console.log(
+      `\n💹 Median ${queryIdentifier} price @ ${queryTime} = ${Number(fromWei(queryPrice.toString())).toFixed(
```

Comment:
`toFixed` introduces a slight rounding error when scaling the price identifier. As an example, a return value of `.086` is changed to `0.085999999999999993`. Any idea why this is?

Reply:
I suspect this is because your `precisionToUse` is set to 18. normally this value is closer to 5. you might need to do the `toFixed` logic differently seeing that javascript does not work well with this kind of tiny number.

One option would be to use the native bignumber `.toFixed` which will properly work. ([docs](https://mikemcl.github.io/bignumber.js/#toFix)). To do this try out something like:

```
toBN(fromWei(queryPrice.toString())).toFixed(precisionToUse)
```

where you convert to a BN rather than to a `Number`.

Reply:
Cool, thank you. I addressed with

`BigNumber(fromWei(queryPrice.toString())).toFixed(precisionToUse)`